docs/ref/models/database-functions.txt

```diff
@@ -1447,6 +1447,84 @@ Usage example::
     >>> print(author.name_code_point)
     77
 
+``RegexpReplace``, ``RegexpStrIndex``, and ``RegexpSubstr``
+-----------------------------------------------------------
+
+.. versionadded:: 4.0
+
+.. class:: RegexpReplace(expression, pattern, replacement=Value(''), flags=Value(''), **extra)
+
+Replaces matches of ``pattern`` with ``replacement`` in ``expression``. The
+default replacement text is the empty string.
+
+.. class:: RegexpStrIndex(expression, pattern, flags=Value(''), **extra)
+
+Returns the offset of the first substring from ``expression`` that matches
+``pattern``.
+
+.. class:: RegexpSubstr(expression, pattern, flags=Value(''), **extra)
+
+Returns the first substring from ``expression`` that matches ``pattern``.
+
+A string of ``flags`` can be provided to adjust the matching and replacement
+behavior for all of the above functions:
+
+* ``c``: Perform case-sensitive matching of ``pattern``.
```

Comment:
I got to this bit and now I'm wondering what the defaults are?

Reply:
The defaults are in theory that no flags are passed. Hence the `flags=Value('')` in the function signatures above.

In reality, it's very messy because nearly none of the backends agree on how to do regular expressions. So we smooth out some of the discrepancies between them to get it as consistent as possible. For PostgreSQL we pass `p` by default, for MySQL we pass `c`, and for MariaDB we use `(?-i)` inline. This is all carefully explained in the admonition.

I wish it were easier, but I believe that this is the best we can do. 🙁